main.tf

```diff
@@ -152,3 +152,11 @@ resource "aws_cloudwatch_metric_alarm" "default" {
     "${null_resource.check_alarm_action.triggers.action}",
   ]
 }
+
+resource "null_resource" "eip" {
+  count = "${signum(length(var.associate_public_ip_address)) == 1 ? 1 : 0}"
+
+  triggers {
+    dns = "ec2-${replace(aws_eip.default.public_ip, ".", "-")}-${data.aws_region.default.name}.compute.amazonaws.com"
```

Comment:
Naming convention is not uniform. `us-east-1` (first region in AWS) uses a slightly different naming scheme.

```
variable "ec2_dns_zones" {
  default = {
    us-west-1 = "us-west-1.compute.amazonaws.com" # e.g. ec2-52-8-166-154.us-west-1.compute.amazonaws.com
    us-west-2 = "us-west-2.compute.amazonaws.com" # e.g. ec2-52-24-51-100.us-west-2.compute.amazonaws.com
    us-east-1 = "compute-1.amazonaws.com"         # e.g. ec2-52-202-45-161.compute-1.amazonaws.com
    eu-west-1 = "eu-west-1.compute.amazonaws.com" # e.g. ec2-52-30-182-167.eu-west-1.compute.amazonaws.com
  }
}
```

Reply:
I think what might work instead is to do what you do in line 160, and replace `us-east-1` with `compute-1`, rather than using a `map` variable.

Reply:
@comeanother recommends using a ternary. I agree.